src/dict/hunspell.cc

```diff
@@ -464,7 +464,7 @@ void HunspellPrefixMatchRequest::run()
 
     QMutexLocker _( &hunspellMutex );
 
-    if ( hunspell.spell( Iconv::toUtf8( Text::utf32, trimmedWord.data(), trimmedWord.size() ) ) ) {
+    if ( hunspell.spell( Iconv::toUtf8( Text::utf32_le, trimmedWord ) ) ) {
```

Comment:
why utf32_le?

Reply:
The direct reason is that `GdWchar` was defined as utf32-le

https://github.com/xiaoyifang/goldendict-ng/blob/c36e008d7d9b3a3362f272537972ea2f7e43484b/src/common/iconv.cc#L9

The deeper reason might be the encode/decode here assumes utf-32 will be little-endian.

https://github.com/xiaoyifang/goldendict-ng/blob/ad192634235a89c62fc07eb766cc180c1533b418/src/common/text.cc#L68-L198

Reply:
It appears that UTF-32 strings like UTF-16 also requires BOM or something similar to properly understand them.

Reply:
IMO, the data itself should not have the BOM mark, or the size should be wrong.

Reply:
UTF32's BOM is 32bits. BOM in std::u32string won't affect the correctness of size().